### ISSUE TITLE
Derive from Serde?

### ISSUE THREAD

**Issue by matthiasbeyer** [soywod/himalaya#153]
Is there any particular reason why you implement `Serialize` and `Deserialize` yourself instead of deriving it?

If no, I would target this as a next refactoring after #151 because it would yield _huge_ benefits (remove a lot of code).
Of course, there should be tests beforehand, to guarantee that the derived implementation is equal to the hand-written one (I could add those as well).

**Comment by soywod** [comment]
Which part of the code are you referring to? If you talk about `Flag` for example, I needed to implement the Serialize trait myself because the `imap::types::Flag` type does not. So I created a wrapper around it. If you have a better idea feel free to propose!

**Comment by matthiasbeyer** [comment]
No, I mean all other impls:

```
$ rg "impl.* Serialize for"
src/flag/model.rs
10:impl<'f> Serialize for SerializableFlag<'f> {
74:impl<'f> Serialize for Flags<'f> {

src/msg/model.rs
41:impl Serialize for Tpl {
115:impl Serialize for ReadableMsg {
195:impl<'a> Serialize for Msg<'a> {
```

**Comment by soywod** [comment]
`SerializableFlag` is the one I was talking about. `Flags` may be removed I guess. For `Tpl` I think I wanted to have a different implementation between Rust model and JSON output, but it can be merged somehow. `ReadableMsg` is clearly useless (I tried without and it works), I guess it is a trash from previous refactor. And `Msg` I think I needed to add 2 fields (`subject_len` and `sender_len`), if there is a better way I'm open. Thanks again for your analysis!
